### PR TITLE
Installer for codegen ui

### PR DESCRIPTION
**Pull request by ashutosh16**

### CODE REVIEW

File: ui/src/ephemeral-access.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ export const RequestAccessBtnFlyout = ({ application }: RequestAccessBtnFlyoutPr
     if (!application) return;
 
     const fetchUserInfo = async () => {
-      const info = await getUserInfo(application);
+      const info = await  getUserInfo(application);
```
Comment (leoluz) [inline ui/src/ephemeral-access.tsx:34]: please remove the unnecessary change in this file

File: Makefile [hunk 2]
```diff
@@ -259,6 +259,15 @@ generate-mocks: mockery ## Generate the mocks for the project as configured in .
 clean-ui:
 	find ${UI_DIR} -type f -name extension.tar -delete
 
+PHONY: codegen-download-openapi
+codegen-download-openapi: run-backend
+	@echo "Downloading OpenAPI spec from dev server"
+	yarn --cwd ${UI_DIR}  api:download
+
+.PHONY: codegen-ui
+codegen-ui: ## Generate the UI API files.
```
Comment (leoluz) [inline Makefile:268]: it seems that this target is dependant on the `build-ui` one as I was getting the following error:
```
❯ make codegen-ui                                                                                   
yarn --cwd /Users/lalmeida1/dev/git/intuit/argocd-ephemeral-access/ui  api:generate
yarn run v1.22.22
warning package.json: No license field
$ orval
/bin/sh: orval: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [codegen-ui] Error 127
```

 If so, please add the `build-ui` as a dependency for this target

Comment (leoluz) [reply Makefile:268]: After running the following commands:
```
make build-ui
make codegen-ui
```
I am getting the following error:
```
❯ make codegen-ui                                                                                              
yarn --cwd /Users/lalmeida1/dev/git/intuit/argocd-ephemeral-access/ui  api:generate
yarn run v1.22.22
warning package.json: No license field
$ orval
🍻 Start orval v7.1.1 - A swagger client generator for typescript
File not found => orval.config.{js,mjs,ts}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [codegen-ui] Error 1
```

It seems that this `orval.config.{js,mjs,ts}` file needs to exist somewhere

File: Makefile [hunk 3]
```diff
@@ -259,6 +259,15 @@ generate-mocks: mockery ## Generate the mocks for the project as configured in .
 clean-ui:
 	find ${UI_DIR} -type f -name extension.tar -delete
 
+PHONY: codegen-download-openapi
+codegen-download-openapi: run-backend
```
Comment (leoluz) [inline Makefile:263]: This doesn't work as it blocks during `run-backend` and never executes the `yarn` command below

File: Makefile [hunk 4]
```diff
@@ -259,6 +259,15 @@ generate-mocks: mockery ## Generate the mocks for the project as configured in .
...
+	@echo "Downloading OpenAPI spec from dev server"
+	yarn --cwd ${UI_DIR}  api:download
```
Comment (leoluz) [inline Makefile:265]: I ran the backend in a separate window and executed the `make codegen-download-openapi`. It deleted the whole `ui/src/gen/schema.yaml` file and updated it with the following line:
```
404 page not found
```

**Review by leoluz — CHANGES_REQUESTED** [review]
Please fix the broken functionality

File: Makefile (hunk 2 above)
Comment (ashutosh16) [reply Makefile:277]: add codegen-ui as prerequisite for build-ui

File: Makefile (hunk 3 above)
Comment (ashutosh16) [reply Makefile]: add the missing files; ran in local and able to generate the schema.yaml

File: Makefile [hunk 5]
```diff
@@ -259,8 +263,22 @@ generate-mocks: mockery ## Generate the mocks for the project as configured in .
 clean-ui:
 	find ${UI_DIR} -type f -name extension.tar -delete
 
+.PHONY: codegen-download-openapi run-backend
+
+codegen-download-openapi:
```
Comment (leoluz) [inline Makefile]: please add `build` and `goreman` as a dependency of this target. Also add the documentation for the target. 
Use this as a reference:
https://github.com/argoproj-labs/argocd-ephemeral-access/blob/d2735f179f10db23ebc7a19269b5c8e02e69795d/Makefile#L106

File: Makefile [hunk 6]
```diff
@@ -259,8 +263,22 @@ generate-mocks: mockery ## Generate the mocks for the project as configured in .
 clean-ui:
 	find ${UI_DIR} -type f -name extension.tar -delete
 
+.PHONY: codegen-download-openapi run-backend
+
```
Comment (leoluz) [inline Makefile:267]: pls remove this empty line

File: Makefile [hunk 7]
```diff
@@ -259,8 +263,22 @@ generate-mocks: mockery ## Generate the mocks for the project as configured in .
 clean-ui:
 	find ${UI_DIR} -type f -name extension.tar -delete
 
+.PHONY: codegen-download-openapi run-backend
```
Comment (leoluz) [inline Makefile:266]: pls fix the `.PHONY` statement. It should only contain the name of the target. 
Ex:
`.PHONY: codegen-download-openapi`

File: Makefile [hunk 8]
```diff
@@ -259,8 +263,22 @@ generate-mocks: mockery ## Generate the mocks for the project as configured in .
...
+
+codegen-download-openapi:  build goreman
```
Comment (leoluz) [inline Makefile:268]: Pls add a brief description about what the target does

**Review by leoluz — CHANGES_REQUESTED** [review]
Pls check my comments

File: Makefile [hunk 9]
```diff
@@ -259,8 +263,21 @@ generate-mocks: mockery ## Generate the mocks for the project as configured in .
...
+	yarn --cwd ${UI_DIR} api:download
+	killall goreman || true
+
+.PHONY: codegen-ui
+codegen-ui: ## Generate the UI API files.
+	yarn --cwd ${UI_DIR}  api:generate
+
 .PHONY: build-ui
-build-ui: clean-ui
+build-ui: clean-ui codegen-ui
```
Comment (leoluz) [inline Makefile:280]: build-ui can not depend on `codegen-ui` as `orval` would never be installed in this case

**Review by leoluz — CHANGES_REQUESTED** [review]

**Review by leoluz — APPROVED** [review]
LGTM
